rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/DiscoverEmbeddedHibernateLibraryRuleProvider.java

```diff
@@ -0,0 +1,84 @@
+package org.jboss.windup.rules.apps.javaee.rules;
+
+import org.jboss.windup.config.AbstractRuleProvider;
+import org.jboss.windup.config.GraphRewrite;
+import org.jboss.windup.config.loader.RuleLoaderContext;
+import org.jboss.windup.config.metadata.RuleMetadata;
+import org.jboss.windup.config.operation.iteration.AbstractIterationOperation;
+import org.jboss.windup.config.phase.InitialAnalysisPhase;
+import org.jboss.windup.config.query.Query;
+import org.jboss.windup.config.query.QueryPropertyComparisonType;
+import org.jboss.windup.graph.GraphContext;
+import org.jboss.windup.graph.model.LinkModel;
+import org.jboss.windup.graph.model.resource.FileModel;
+import org.jboss.windup.graph.service.LinkService;
+import org.jboss.windup.reporting.category.IssueCategoryRegistry;
+import org.jboss.windup.reporting.model.ClassificationModel;
+import org.jboss.windup.reporting.model.TechnologyTagLevel;
+import org.jboss.windup.reporting.service.ClassificationService;
+import org.jboss.windup.reporting.service.TechnologyTagService;
+import org.jboss.windup.rules.apps.java.model.JarArchiveModel;
+import org.ocpsoft.rewrite.config.Configuration;
+import org.ocpsoft.rewrite.config.ConfigurationBuilder;
+import org.ocpsoft.rewrite.context.EvaluationContext;
+
+@RuleMetadata(phase = InitialAnalysisPhase.class, perform = "Discover Java libraries embedded")
+public class DiscoverEmbeddedHibernateLibraryRuleProvider extends AbstractRuleProvider
+{
+
+    @Override
+    public Configuration getConfiguration(RuleLoaderContext context)
+    {
+        String ruleIDPrefix = getClass().getSimpleName();
+        int ruleIDSuffix = 1;
+        return ConfigurationBuilder.begin()
+                    .addRule()
+                    .when(Query.fromType(JarArchiveModel.class)
+                                .withProperty(FileModel.FILE_PATH, QueryPropertyComparisonType.REGEX, ".*WEB-INF/lib/.*hibernate.*\\.jar$"))
```

Comment:
I realize that introduces a small possibility that they might have them somewhere in the archive that is not actually picked up by the application server. That seems like a small risk, though, compared to potentially missing competitive server's preferred location (I think some use APP-LIB or something like that).